service/smithyprototype/lexruntimeservice/api_types.go

```diff
@@ -487,3 +488,123 @@ func (s SentimentResponse) MarshalFields(e protocol.FieldEncoder) error {
 	}
 	return nil
 }
+
+func serializeIntentSummaryListAWSJSON(v []IntentSummary, value json.Value) error {
+	if v == nil {
+		value.Null()
+		return nil
+	}
+
```

Comment:
Agreed, though in this context serializeIntentSummaryListAWSJSON would not have been called if the list is nil. So this was just more of a fallback case if the intent was to actually encode a null. Otherwise this would have to return an error to signal it can't populate a null value.

Reply:
What do you think about using error as the return behavior in this case?  Since we're not actively supporting encoding null? Would prevent unexpected code paths silently encoding null when its not supposed to be supported.